### ISSUE TITLE
mini.starter - segfault when refresh

### ISSUE THREAD

**Issue by soifou** [echasnovski/mini.nvim#147]
### Contributing guidelines

- [X] I have read [CONTRIBUTING.md](https://github.com/echasnovski/mini.nvim/blob/main/CONTRIBUTING.md)
- [X] I have read [CODE_OF_CONDUCT.md](https://github.com/echasnovski/mini.nvim/blob/main/CODE_OF_CONDUCT.md)
- [X] I have updated 'mini.nvim' to latest version

### Module(s)

mini.starter

### Description

First thank you for your hard work on mini. I started to use it 8 months ago for replacement of surround and from time to time I like to check your other modules ; pairs, cursorword and starter are now in my toolbox and probably others in the future :)

To get back on this bug, this is probably a very edge/unsual case but since it crashes nvim I prefer to report!

Some terminals like [kitty](https://sw.kovidgoyal.net/kitty/conf/#opt-kitty.window_padding_width) or [wezterm](https://wezfurlong.org/wezterm/config/lua/config/window_padding.html?highlight=padding#window_padding) allow to set a padding that can be modified at runtime with user defined keybinds. I usually reset my default padding to 0 when I'm working with nvim to get the maximum space.

I notice whenever `mini.starter` is displayed and focused if I reset this padding, nvim crashes, arguing:

```
double free or corruption (!prev)
zsh: segmentation fault (core dumped)
```

I tested on zsh and bash, with kitty and wezterm, same behaviour.

This is not happening if I do this operation on other buffers, even if starter is displayed.
My workaround for now is to remove `MiniStarterBuffer` augroup when hooked on `MiniStarterOpened`
So, I might be wrong but I suspect `Mini.refresh()` to be the culprit. 

Last (and weird) thing, this is not happening if I had more padding (!?) 

### Neovim version

NVIM v0.9.0-dev-128-gef4c339fb

### Steps to reproduce

```sh
$ kitty -o allow_remote_control=yes -o window_padding_width=20 -o map="alt+0 remote_control set-spacing padding=0" nvim -nu minimal.lua
```

Content of minimal.lua

```lua
vim.cmd([[set runtimepath=$VIMRUNTIME]])
vim.cmd([[set packpath=/tmp/nvim/site]])

local package_root = '/tmp/nvim/site/pack'
local packer_install_path = package_root .. '/packer/start/packer.nvim'

local function load_plugins()
    require('packer').startup({
        {
            'wbthomason/packer.nvim',
            'echasnovski/mini.nvim',
        },
        config = {
            package_root = package_root,
            compile_path = packer_install_path .. '/plugin/packer_compiled.lua',
            display = { non_interactive = true },
        },
    })
end

_G.load_config = function()
    require('mini.starter').setup()
    require('mini.starter').open()
end

if vim.fn.isdirectory(packer_install_path) == 0 then
    print('Installing plugins and dependencies...')
    vim.fn.system({ 'git', 'clone', '--depth=1', 'https://github.com/wbthomason/packer.nvim', packer_install_path })
end

load_plugins()
require('packer').sync()
vim.cmd( [[autocmd User PackerComplete ++once echo "Ready!" | lua load_config()]])
```

Wait for mini.starter to display. Then hit `alt+0` to reset padding.

If you invert the values it works:

```sh
$ kitty -o allow_remote_control=yes -o window_padding_width=0 -o map="alt+0 remote_control set-spacing padding=20" nvim -nu minimal.lua
```

### Expected behavior

nvim should not crash

### Actual behavior

nvim crashes

**Comment by echasnovski** [comment]
Thanks for kind words and an interesting debugging problem.

I am able to reproduce this but only in development version of Neovim (`NVIM v0.9.0-dev-109-g0b71960ab` in my case). Neovim 0.8.0 release works fine. This indeed happens because `MiniStarter.refresh()` is called on `VimResized` event.

Not yet sure what caused this, but my initial feeling is that the problem is on Neovim side. I'll look into it.

**Comment by echasnovski** [comment]
I am positive that this is a result of neovim/neovim#20759.

If it bothers you too much, you can replace [these lines](https://github.com/echasnovski/mini.nvim/blob/3202e7a298f4fe7513b4dad409075e6b618488f9/lua/mini/starter.lua#L1271) with this:
```
au VimResized <buffer=%s> lua vim.schedule(MiniStarter.refresh)
```

I'll wait to see how Neovim issue is resolved. If it won't be fixed, I'll update 'mini.starter' to echo on in scheduled manner.

**Comment by soifou** [comment]
Oh now you say it, this is indeed likely a core issue and did not experienced this problem some weeks ago. I can live with the disabled augroup for now  (there is already a PR that fix the issue, that was fast!)

Not related but since we're talking about `MiniStarter.refresh` I notice refreshing from another buffer like a treeview or help pane, `:lua MiniStarter.refresh(buf_mini_starter_id)` won't correctly align, if I issue the exact same command when it is focused, this works. am I missing something?

**Comment by echasnovski** [comment]
> Not related but since we're talking about `MiniStarter.refresh` I notice refreshing from another buffer like a treeview or help pane, `:lua MiniStarter.refresh(buf_mini_starter_id)` won't correctly align, if I issue the exact same command when it is focused, this works. am I missing something?

No, you don't miss anything. This is a bug because aligning hook currently [gets width from current window](https://github.com/echasnovski/mini.nvim/blob/a57bd4bc25bedd8c0c7d27899f86fa909d157752/lua/mini/starter.lua#L716). This is a part I overlooked when implementing #82 - previously it was only single Starter buffer allowed.

I'll fix it. Thanks again for noticing!

**Comment by soifou** [comment]
> I'll fix it. Thanks again for noticing!

No problem! Do you want I create a specific issue for this?

**Comment by echasnovski** [comment]
> > I'll fix it. Thanks again for noticing!
> 
> No problem! Do you want I create a specific issue for this?

No, thanks. I am already on it.

**Comment by echasnovski** [comment]
Calling `MiniStarter.refresh()` when not inside Starter buffer should be fixed in latest `main`. Could you, please, test it?

**Comment by soifou** [comment]
Yes I confirm this fixed it! Thank you!

**Comment by soifou** [comment]
[Neovim PR #20760](https://github.com/neovim/neovim/pull/20760) has been merged, rebuilded from head (`v0.9.0-dev-150+g2f9b94a26`) and no more segfault :) 

Problem solved!
